sdk/src/client/node_manager/builder.rs

```diff
@@ -87,45 +77,36 @@ impl NodeManagerBuilder {
             url,
             auth: None,
             disabled: false,
+            permanode: false,
         }));
         Ok(self)
     }
 
-    pub(crate) fn with_primary_node(mut self, url: &str, auth: Option<NodeAuth>) -> Result<Self> {
-        let mut url = validate_url(Url::parse(url)?)?;
-        if let Some(auth) = &auth {
-            if let Some((name, password)) = &auth.basic_auth_name_pwd {
-                url.set_username(name)
-                    .map_err(|_| crate::client::Error::UrlAuth("username"))?;
-                url.set_password(Some(password))
-                    .map_err(|_| crate::client::Error::UrlAuth("password"))?;
-            }
+    pub(crate) fn with_primary_nodes(mut self, urls: &[&str]) -> Result<Self> {
+        for url in urls {
+            let url = validate_url(Url::parse(url)?)?;
+            self.primary_nodes.push(NodeDto::Node(Node {
+                url,
+                auth: None,
+                disabled: false,
+                permanode: false,
+            }));
         }
-        self.primary_node.replace(NodeDto::Node(Node {
-            url,
-            auth,
-            disabled: false,
-        }));
         Ok(self)
     }
 
-    pub(crate) fn with_permanode(mut self, url: &str, auth: impl Into<Option<NodeAuth>>) -> Result<Self> {
-        let mut url = validate_url(Url::parse(url)?)?;
-        let auth = auth.into();
-        if let Some(auth) = &auth {
+    pub(crate) fn with_primary_node(mut self, mut node: Node) -> Result<Self> {
```

Comment:
Ignore if you want, but if you can spare the time you could do a bit of cleanup in this PR and order the methods a bit: I'ld suggest something like `with_node`, `with_nodes`, `with_primary_node`, `with_primary_nodes`.

Reply:
Reordered them, but a bit different